modules/raft/src/main/java/org/apache/ignite/raft/jraft/rpc/impl/RaftGroupServiceImpl.java

```diff
@@ -194,7 +201,7 @@ private RaftGroupServiceImpl(
         sendWithRetry(randomNode(), req, currentTimeMillis() + timeout, fut);
 
         return fut.thenApply(resp -> {
-            leader = resp.leader();
+            leader = PeerId.parsePeer(resp.leaderId());
```

Comment:
Is it possible that resp.leaderId() will be null?

Reply:
Yes, and parsePeer will return null in this case.